### PR TITLE
Migrate from setup.py to pyproject.toml and Poetry

### PR DESCRIPTION
**Pull request by FuzzyReality**
We have decided to move from `setup.py` to poetry and `pyproject.toml` as our python package and dependency management tool. This is mainly done in order to ease the integration of SNYK. SNYK is used for vulnerability scanning of our dependencies.

As part of this, relevant workflows have also been updated to work with poetry.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -7,18 +7,19 @@ RUN python -m venv --copies $VIRTUAL_ENV
 ENV PATH="$VIRTUAL_ENV/bin:$PATH"
 
 RUN python -m pip install --upgrade pip
+RUN pip install poetry==1.7.0
```
Comment (Christdej) [inline Dockerfile:10]: Is it necessary to lock the version of this?

**Review by Christdej — CHANGES_REQUESTED** [review]

File: .github/workflows/pythonpackage.yml [hunk 2]
```diff
@@ -14,28 +14,32 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.10", "3.11", "3.12"]
+        python-version: ["3.10", "3.11", "3.12", "3.x"] # pythonpublish.yml uses "3.x".
```
[the remark this answers is not in the document]
Comment (FuzzyReality) [reply .github/workflows/pythonpackage.yml:17]: I'm fine with that. Added `3.x` as pythonpublish.yml was (and still is) using `3.x`

File: .github/workflows/pythonpublish.yml [hunk 3]
```diff
@@ -1,24 +1,57 @@
 name: Upload Python Package
 
+# This is an adaptation of https://github.com/code-specialist/pypi-poetry-publish.
+# That action is released under an MIT License, and as such, so is this workflow.
```
[the remark this answers is not in the document]
Comment (FuzzyReality) [reply .github/workflows/pythonpublish.yml:4]: Maybe, maybe not. ISAR uses EPLv2, while the one I based my work on uses MIT. Don't feel I've changed it sufficiently from the original to justify changing the License. (And I've also been trained to reference my sources ;) )
I'd be happy to discuss this.

File: Dockerfile (hunk 1 above)
Comment (FuzzyReality) [reply Dockerfile:10]: No. It is recommended practice though.

File: src/isar/__init__.py [hunk 4]
```diff
@@ -1,6 +1 @@
-from pkg_resources import DistributionNotFound, get_distribution
-
-try:
-    __version__ = get_distribution(__name__).version
-except DistributionNotFound:
-    pass  # package is not installed
+__version__ = "1.16.8"
```
[the remark this answers is not in the document]
Comment (Christdej) [reply src/isar/__init__.py:1]: Could just be removed altogether or not?

**Comment by aestene** [comment]
This pull request has automatically been marked as stale as there has been no activity for 30 days.

**Comment by tsundvoll** [comment]
I suggest we migrate from setup.py to pyproject.toml as proposed by this PR as well, but that we stick to using pip and setuptools for packaging and building. A have made a PR on isar-robot with my suggested solution: https://github.com/equinor/isar-robot/pull/103. Something similar can be done for ISAR if we decide to to that way.

File: .github/workflows/pythonpublish.yml [hunk 5]
```diff
@@ -1,24 +1,57 @@
...
-        uses: actions/setup-python@v3
+        uses: actions/setup-python@v4
         with:
           python-version: "3.x"
+
+      - name: Set GitHub Tag as Package Version
+        run: |
+          sed -i -r 's/__version__ *= *".*"/__version__ = "${{ github.event.release.tag_name }}"/g' ./src/isar/__init__.py
+          sed -i '0,/version =.*/s//version = "'"${{ github.event.release.tag_name }}"'"/' ./pyproject.toml
+        shell: bash
```
Comment (tsundvoll) [inline .github/workflows/pythonpublish.yml:31]: If we stick to setuptools, we can use the SCM tag (github release tag) for setting the version and we avoid having a separate version bump commit.

See example here: https://github.com/equinor/isar-robot/pull/103/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R34-R35

**Pull request closed without merging**